### DISCUSSION TITLE
Python: modify response body

### DISCUSSION THREAD

**Discussion by CedoAlteram** [mitmproxy/mitmproxy#5282] [Q&A] [answered]
Hey, 

Thanks for this proj, it's awesome to see something like this. 

I'm looking for a way to modify the response body so that I have malformed json. 

Ex. 
`curl wtfismyip.com/json`

should return something like

```{
    "YourF******IPAddress": "1.2.3.4",
    "YourF******Location": "Detroit, MI, United States",
    "YourF******Hostname": "1-2-3-4.ptld.qwest.net",
    "YourF******ISP": "CenturyLink",
    "YourF******TorExit": false,
    "YourF******CountryCode": "US"
}
```

However, I want non-valid json, which means I want to be able to catch the response and malform the json (on purpose). Testing something.

I'm looking at addons and I can see the headers, but Im wondering how to see the response body and how to edit it. 

```from mitmproxy import ctx


class Counter:
    def __init__(self):
        self.num = 0

    def request(self, flow):
        self.num = self.num + 1
        ctx.log.info(f"We've seen {self.num} flows, count is {self.num*2}")

    def response(self, flow):
        print(flow.response.headers)
        print(flow.response)
        ctx.log(f"response: {flow=}")


addons = [
    Counter()
]
```

Any pointers as to where to look would be appreciated!

a

**Comment by Prinzhorn** [comment, marked as answer]
https://docs.mitmproxy.org/stable/api/events.html#HTTPEvents.response
https://docs.mitmproxy.org/stable/api/mitmproxy/http.html#HTTPFlow
https://docs.mitmproxy.org/stable/api/mitmproxy/http.html#Response
https://docs.mitmproxy.org/stable/api/mitmproxy/http.html#Message.text
https://docs.mitmproxy.org/stable/addons-examples/#internet-in-mirror

E.g. `flow.response.text = 'invalid json'` or `flow.response.content = b'invalid json'`

**Comment by CedoAlteram** [comment]
Ah! that makes sense! Thanks @Prinzhorn

**Reply by rosydawn6** [reply]
I found this and a few other links looking for information on this topic.  Several attempts were not producing the expected results.  I was using stream_large_bodies=1 and after rereading the docs more https://docs.mitmproxy.org/stable/concepts-options/#stream_large_bodies found that when using stream_large_bodies=1 the body is not stored and hence cannot be modified.
